pkg/plugins/golang/v3/scaffolds/init.go

```diff
@@ -112,5 +113,12 @@ func (s *initScaffolder) Scaffold() error {
 		},
 		&templates.Dockerfile{},
 		&templates.DockerIgnore{},
+		&templates.Readme{
+			BoilerplateMixin: machinery.BoilerplateMixin{
```

Comment:
Why we are passing that and not using it in the boilerplate itself as we do in the other similar scenarios?

Reply:
Could we try to as we do in the other places keep the same standard and let the boilerplate itself have the code and the logic that is required to build it?

Reply:
I was having some difficulty understanding this part but understood it much better after reading some of your other comments regarding the boilerplate. 

I made changes in commit https://github.com/kubernetes-sigs/kubebuilder/pull/2610/commits/a0faa6f08895abaf1bad432af586e49ef8ce107e

Let me know what you think